docs/design/storagenode-graceful-exit-pieces.md

```diff
@@ -0,0 +1,42 @@
+# Storage Node Graceful Exit - Gathering Pieces
+
+## Abstract
+
+During [Graceful Exit](storagenode-graceful-exit-overview.md), a satellite needs to find pieces to be transferred from the exiting storage node.
+
+## Background
+
+Graceful Exit contains a process that moves existing pieces from one storage node to another. To accomplish this we need a list of pieces that need to be transferred.
+
+Pieces with lower durability have higher importance to be transferred.
+
+## Design
+
+We need a service on the satellite that finds pieces in the metainfo database that need to be transferred. We'll call this service `gexit.Service` or Graceful Exit service.
+
+The service starts by asking overlay for all exiting nodes.
+
+Then joins a metainfo loop to iterate over all segments. For any segment that contains nodes that are exiting it will add an entry to a queue. We call this transfer queue.
+
+The transfer queue is stored in database. We will need batching when inserting to database to avoid excessive load.
+
+Once metainfo loop has completed successfully it updates node to be ready for transferring.
+
+## Rationale
+
+We could store the queue in-memory, however there is a danger that it might consume too much memory. We can simplify the queue, by not having batching, however this would significantly increase the database load.
+
+We could keep keep a live summary of the pieces in the queue, however, we can always query the database, which is easier to implement and change.
+
+The metainfo loop `Join` guarantees the observer will only receive events at the beginning of the next loop. Hence, one complete metainfo loop is sufficient to collect all the pieces for a given node. 
+
+## Implementation
+
+1. Add method for finding exiting nodes to overlay.
+2. Implement transfer queue for pieces.
+3. Implement gexit.Service.
+4. Update satellite to ignore exiting storage nodes for repairs and uploads.
+
+## Open issues (if applicable)
+
+- Can pieces with really high durability can be ignored?
```

Comment:
I think this is against the economics of the graceful exit. We want the exiting nodes to pay the price for future repairs due to leaving the network. Even if a piece has high durability, it still makes sense to transfer it to a new node. Otherwise, the durability may degrade over time due to gracefully exiting nodes, which will trigger the repair sooner, which will increase the cost on the satellite. A successful graceful exit should not have any negative impact on durability.

Reply:
If this is the case then it makes sense. However, I'd expect this to be quite a rare case. Does the monitoring currently report any segment with such high durability?

Reply:
OK. I agree then.

Reply:
Updated to remove the exiting node when durability > optimal.